src/include/network/lan/address.rb

```diff
@@ -1631,5 +1611,47 @@ def repeated_physical_port_ids(slaves)
 
       physical_port_ids
     end
+
+    # Performs hostname update
+    #
+    # This handles ip and hostname change when editing NIC properties.
+    # The method relies on old NIC's IP which is set globally at initialization
+    # of NIC edit dialog (@see LanItems#ipaddr)
+    #
+    # When hostname is empty, then old IP's record is cleared from /etc/hosts and
+    # new is not created.
+    # When hostname is changed then canonical name and all aliases in the record
+    # are replaced by new ones.
```

Comment:
OK, but then this test case becomes misleading, because it shows a variant where aliases are preserved, but the code is never called that way. Is that intentional?
https://github.com/yast/yast-network/blob/a3e23d6af35defe2ae61e9aae8cb8ef5302d5de3/test/host_test.rb#L237;L245

Reply:
that test testes behavior under the hood - the old API. In this PR I've tried to create a new method which should handle the behavior in a defined way. Of course that it is possible to do more with underlying API.